crowbar_framework/app/helpers/nodes_helper.rb

```diff
@@ -88,16 +88,22 @@ def nodes_hash(group = nil)
     @nodes_hash[group] ||= begin
       {}.tap do |nodes|
         NodeObject.all.each do |node|
-          nodes[node.name] = {
-            :handle => node.handle,
-            :alias => node.alias,
-            :title => node.description(false, true),
-            :admin => node.admin?,
-            :group => node.group,
-            :platform => node[:platform],
-            :platform_version => node[:platform_version],
-            :cluster => false
-          } if node.group == group or group.nil?
+          if node.group == group || group.nil?
+            nodes[node.name] = {
+              :handle => node.handle,
+              :alias => node.alias,
+              :title => node.description(false, true),
+              :admin => node.admin?,
+              :group => node.group,
+              :platform => node[:platform],
+              :platform_version => node[:platform_version],
+              :cluster => false
+            }
+            if node["crowbar_wall"] && node["crowbar_wall"]["disk_roles"]
```

Comment:
I would agree with @tboerger; I understand that the goal is to avoid running code uselessly, but the way it reads is a bit convoluted. Maybe we should rethink the changes you did in node_object.rb and make the disk_roles method just return the attribute, and add a new `refresh_disk_roles` method?